### PR TITLE
chore: Update @metamask/delegation-controller from `^2.0.2` to `^3.0.0`

### PR DESCRIPTION
**Pull request by jeffsmale90**
## **Description**

Updates @metamask/delegation-controller to [3.0.0](https://github.com/MetaMask/core/blob/main/packages/delegation-controller/CHANGELOG.md#300):
> BREAKING: Remove persisted delegations state (https://github.com/MetaMask/core/pull/8330)
>   - store, list, retrieve, chain, and delete methods (and related messenger action types)
>   - DelegationEntry type export
>   - Remove dependency on @metamask/accounts-controller - callers no longer need to delegate the  AccountsController:getSelectedAccount action to the DelegationController messenger

Removes related cruft from extension codebase required to initialise and manage stored permissions.

Because the delegation-controller's storage was never used, there's no need to manage stored data, aside from migration 208 that simply removes `delegation` key.

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

No functional change - this removes unused code.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> **Medium Risk**
> Moderate risk due to upgrading a core controller dependency and removing multiple background/UI APIs; failures would primarily affect delegation-related flows and persisted state migration, not broader wallet functionality.
> 
> **Overview**
> Updates `@metamask/delegation-controller` to `^3.0.0` and aligns extension integration by **dropping all delegation-entry persistence APIs** (e.g., `store/list/retrieve/chain/delete` and `awaitDeleteDelegationEntry`) from the background init layer and UI controller actions, leaving only `signDelegation` exposed.
> 
> Simplifies the delegation messenger wiring by removing delegated AccountsController actions and all TransactionController event subscriptions, and deletes the now-unused UI selectors/tests for reading `metamask.delegations`.
> 
> Adds migration `210` (and fixture/version bumps) to remove any legacy persisted `DelegationController.delegations` data from disk, with unit tests covering common edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4b1c2645854db40eca583bdd3f5fd68d89124f47. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by jeffsmale90** [comment]
@metamaskbot update-e2e-fixture

**Comment by jeffsmale90** [comment]
@metamaskbot update-e2e-fixture

**Review by GuillaumeRx — COMMENTED** [review]
LGTM ! Although I'm not a codeowner on any of these files 😄

File: test/e2e/tests/solana/mocks/securityAlertSwap.json [hunk 1]
```diff
@@ -97,7 +97,6 @@
           }
         ]
       },
-      "delegations": {},
```
[the remark this answers is not in the document]
Comment (gauthierpetetin) [reply test/e2e/tests/solana/mocks/securityAlertSwap.json:100]: Out of curiosity, why is this comment not valid?

**Review by gauthierpetetin — APPROVED** [review]
Just left a non-blocking question

**Review by seaona — APPROVED** [review]
e2e files LGTM

**Review by mj-kiwi — APPROVED** [review]
LGTM, maybe need to run the `yarn lavamoat:auto` to update the attributions file
